CARET_trace/test/test_trace_node.cpp

```diff
@@ -175,8 +195,33 @@ TEST_F(CaretTraceNodeTest, TestPrepareTransition)
     EXPECT_TRUE(node_->get_status() == TRACE_STATUS::PREPARE);
   }
 
-  {  // subscribe end message: PREPARE -> WAIT
+  {  // subscribe start message without lttng session: PREPARE -> PREPARE
+    set_status(TRACE_STATUS::PREPARE);
+    set_lttng_session_running_return_value(false);
+
+    EXPECT_TRUE(node_->get_status() == TRACE_STATUS::PREPARE);
+
+    auto start_msg = get_start_msg();
+    node_->start_callback(std::move(start_msg));
+
+    EXPECT_TRUE(node_->get_status() == TRACE_STATUS::PREPARE);
+  }
+
+  {  // subscribe end message with lttng session: PREPARE -> WAIT
+    set_status(TRACE_STATUS::PREPARE);
+    set_lttng_session_running_return_value(true);
+
+    EXPECT_TRUE(node_->get_status() == TRACE_STATUS::PREPARE);
+
+    auto end_msg = get_end_msg();
+    node_->end_callback(std::move(end_msg));
+
+    EXPECT_TRUE(node_->get_status() == TRACE_STATUS::PREPARE);
+  }
+
+  {  // subscribe end message without lttng session: PREPARE -> PREPARE
```

Comment:
Minor fix of comment.

```suggestion
  {  // subscribe end message without lttng session: PREPARE -> WAIT
```